### DISCUSSION TITLE
Garmin Varia 515 Solid Mode without the Flashing / exclude position to trigger filter

### DISCUSSION THREAD

**Discussion by ghost** [maca88/SmartBikeLights#35] [Q&A]
Hi there,

this app is amazing and i am really happy someone put in the work.

In my country (germany) the flash mode is technically not allowed when mounted on the bike directly (although it is when attached to a bag which is mounted to the bike :O).

I bought the 515 for the flashing mode in combination with smartbikelights (without the app i would have just bought the 516) for my day rides (i have global filter: solid when sunset to sunrise, flash when sunrise to sunset and radar sees car), i noticed that after sunset when solid mode is active the light is still flashing when it sees a car behind. Is there a way to stop the additional flashing in solid mode or is this expected behaviour and cant be deactivated?

And is there a way to exclude gps coordinates to trigger a filter, i tried the "position" filter but as far as i understand it u can only trigger it when inside the triangle. I want my varia to enable day light flash with the 3 filters:
- from sunset to sunrise
- thread operate >= 0
- outside of city

thnx in advance

**Comment by maca88** [comment]
Hi,

>  Is there a way to stop the additional flashing in solid mode or is this expected behaviour and cant be deactivated?

You need to put the sunset/sunrise filter group before the radar one and then the light will stop flashing after sunset. The order of filter groups is important, the light mode of the first group (from top to bottom) that is matched will be used.


> is there a way to exclude gps coordinates to trigger a filter,

Currently it is not, but you can still achieve what you desire by setting the light mode to "Off" when inside the city:
`
1,1!:1H]0###0,73535488#4,4!NIGHT:1:4:0:0Es0,r0!CITY:1:0:0:0F1,48.186669,11.463203,48.186669,11.630058,48.095946,11.638985,48.098239,11.462517!RADAR:1:7:0:0I]-1]0!:1:0:0:0D=1##6,5:Varia 515!2,:-1,Off:0!1,Solid:4!1,Peloton:5!1,Day Flash:7!1,Night Flash:6#0::#0:0#123!:123!#B3843##3#0#0
`

the above configuration will:

1. Turn to Solid after sunset
2. Turn off when inside a city (added Munchen as example)
3. Turn to "Day flash" when in daytime, outside the city and a vehicle is approaching you

**Comment by ghost** [comment]
thnx, munich was exactly the place i was looking for :).

but the solid flash behaviour is still the same,
i used the provided filter example from the docs:
"Taillight with solid mode in the nighttime and flashing mode in the daytime"

Solid Mode is the first group and the filter is triggered as supposed to be after sunset (the light turns solid), but still when radar detects a car the solid light begins to flash and returns to solid mode after the car passed by. (i was very tired when driving so maybe i did something wrong and will test it again, but i am pretty sure it was solid+flashing).
The issue is described here (for the 510....)
https://forums.garmin.com/sports-fitness/cycling/f/varia-series/193011/rtl510-group-rides-solid-mode?pifragment-1290=1

btw ur taking donations?

**Reply by maca88** [reply]
Can you please share the configuration that you are currently using (you can edit the position polygons for privacy before pasting), so that I can test it out? If the "Solid" light mode is the first group then it shouldn't switch to flashing when the "Solid" group is active.

> The issue is described here (for the 510....)

That is new to me that RTL510 is flashing when a vehicle is approaching without using a CIQ application.

> btw ur taking donations?

Currently I don't take donations, maybe I will add an option in the future.

**Reply by ghost** [reply]
I didn’t test your configuration yet, will report back when I have another night tour.

i noticed the solid flash problem using the provided example from the docs:

https://maca88.github.io/SmartBikeLights/?c=IyMjMCw3MzQwNDQxNiMzLDMhTklHSFQ6MTo0OjA6MEVzMCxyMCFEQVk6MTo3OjA6MEVyMCxzMCE6MTowOjA6MEQ9MSMjNSw0OlZhcmlhIDUxMCEyLDotMSxPZmY6MCExLFNvbGlkOjQhMSxEYXkgRmxhc2g6NyExLE5pZ2h0IEZsYXNoOjYjMDo6IzA6MCNCMjcxMyMjMiMwIzA=

I did only add the thread operator >=none for the second day flash filter.(the provided example uses from sunrise to sunset filter which is I think not mandatory)

**Comment by Rossi0209** [comment]
> Hi there,
> 
> this app is amazing and i am really happy someone put in the work.
> 
> In my country (germany) the flash mode is technically not allowed when mounted on the bike directly (although it is when attached to a bag which is mounted to the bike :O).
> 
> I bought the 515 for the flashing mode in combination with smartbikelights (without the app i would have just bought the 516) for my day rides (i have global filter: solid when sunset to sunrise, flash when sunrise to sunset and radar sees car), i noticed that after sunset when solid mode is active the light is still flashing when it sees a car behind. Is there a way to stop the additional flashing in solid mode or is this expected behaviour and cant be deactivated?
> 
> And is there a way to exclude gps coordinates to trigger a filter, i tried the "position" filter but as far as i understand it u can only trigger it when inside the triangle. I want my varia to enable day light flash with the 3 filters:
> 
> * from sunset to sunrise
> * thread operate >= 0
> * outside of city
> 
> thnx in advance

Hi, i have the same issue with an Garmin Edge 1040 and a Varia RCT 715. 

Car is detected, the light flashes, car hast passend through the light is solid.

So you have meanwhile a solution?
Regards
Marcus

**Reply by maca88** [reply]
Hi,

can you share the configuration that you are using?

**Comment by Rossi0209** [comment]
Hi, 

Here is my actual config
###0,73535488:1,1274059773#2,2!CARON:1:5:2:0I[300]0!:1:0:0:0D=1##6,3:Varia 715 DE!2,:-1,Off:0!2,Solid:4,Peloton:5!2,Day Flash:7,Night Flash:6#0::#0:1#123!:123!#B3843##11#0#0

Btw, great App!

**Comment by maca88** [comment]
Your configuration does not have a solid mode, so I suppose you are not using the Smart mode. Make sure that character "S" is displayed inside the light icon or on the top left button in case you have the field set to full screen. The behavior that you described is exactly what the Peloton does by default (without CIQ app), when a vehicle is approaching you the light starts flashing otherwise is solid.

**Reply by Rossi0209** [reply]
Hi, 
i have verified the config an created a complete new from Scratch for my RT510. The issue ist the same. If a car is aproaching, the light blinks. 

I have taken a Video. The Mode is on S and the Light is configured solid. 

Video:
https://youtube.com/shorts/9W_qnK8KT1s?feature=share

Config:
###0,73404416:1,1163788867#2,2!KFZ:1:4:0:0I[300]0!:1:0:0:0D=1##5,4:Varia 510!2,:-1,Off:0!1,Solid:4!1,Day Flash:7!1,Night Flash:6#0::#0:1#123!:123!#B3843##2#0#0

**Reply by maca88** [reply]
Thanks for the video, I tried it myself with RTL515 and indeed I get the same behavior. I didn't know that also the Solid mode (along with Peloton mode) goes automatically to flashing when a vehicle is approaching natively (without CIQ app). Unfortunately this behavior is embedded into the Varia firmware and there is no API that we can use to change the behavior. I do apologize for wasting your time for something that I should already know by now :/
